app/controllers/NewAuthYesNoController.scala

```diff
@@ -16,32 +16,59 @@
 
 package controllers
 
-import controllers.actions._
+import controllers.actions.*
 import forms.YesNoFormProvider
+import models.requests.DataRequest
 import models.{ArrivalId, Mode, UserAnswers}
 import navigation.Navigation
 import pages.NewAuthYesNoPage
 import play.api.i18n.{I18nSupport, MessagesApi}
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import play.api.mvc.Results.Redirect
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
 import repositories.SessionRepository
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import utils.transformers.IE043Transformer
 import views.html.NewAuthYesNoView
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
+trait NewAuthYesNoSubmission {
```

Comment:
Not a bad solution. Did you consider moving the logic out into some kind of service layer class that could be injected into both controllers? Similar result but would mean things like the `SessionRepository` and `IE043Transformer` and any future dependencies related to this common code would only need to be injected once into the service layer class.